src/main/filesystem.ts

```diff
@@ -26,69 +28,88 @@ class LauncherFileSystem {
-    const dataDir = path.join(app.getPath('userData'), 'data');
-
-    console.log('Got logsDir, configDir and dataDir: ', logsDir, configDir, dataDir);
-
-    createDirIfNotExists(logsDir);
-    createDirIfNotExists(configDir);
-    createDirIfNotExists(dataDir);
-
-    const launcherFileSystem = new LauncherFileSystem(dataDir, configDir, logsDir);
+    const launcherFileSystem = new LauncherFileSystem(profileDataDir, profileLogsDir, profile);
 
     launcherFileSystem.createInitialDirectoryStructure();
     return launcherFileSystem;
   }
 
   createInitialDirectoryStructure = () => {
     createDirIfNotExists(this.keystoreDir);
-    createDirIfNotExists(path.join(this.appDataDir, 'holochain'));
-    createDirIfNotExists(path.join(this.appDataDir, 'uis'));
-    createDirIfNotExists(path.join(this.appConfigDir, 'holochain'));
+    createDirIfNotExists(this.holochainDir);
   };
 
   get keystoreDir() {
-    return path.join(this.appDataDir, 'lair');
+    return path.join(this.profileDataDir, 'lair');
   }
 
   get holochainDir() {
-    return path.join(this.appDataDir, 'holochain');
+    return path.join(this.profileDataDir, 'holochain');
+  }
+
+  holochainPartitionDir(partition: string) {
+    return path.join(this.holochainDir, partition);
+  }
+
+  conductorConfigPath(partition: string) {
+    return path.join(this.holochainDir, partition, 'conductor-config.yaml');
   }
 
-  get uisDir() {
-    return path.join(this.appDataDir, 'uis');
+  conductorEnvironmentDir(partition: string) {
+    return path.join(this.holochainDir, partition, 'dbs');
   }
 
-  get conductorConfigPath() {
-    return path.join(this.appConfigDir, 'holochain', 'conductor-config.yaml');
+  happUiDir(appId: string, partition: string) {
+    return path.join(this.holochainPartitionDir(partition), 'apps', appId, 'ui');
   }
 
-  appUiDir(appId: string) {
-    return path.join(this.uisDir, appId);
+  happConfigPath(appId: string, partition: string) {
+    return path.join(this.holochainPartitionDir(partition), 'apps', appId, 'app-config.json');
   }
 
   keystoreInitialized = () => {
     return fs.existsSync(path.join(this.keystoreDir, 'lair-keystore-config.yaml'));
   };
+
+  // TODO
+  // factoryReset = (deleteLogs: boolean) => {};
 }
 
-function createDirIfNotExists(path: fs.PathLike) {
+export function createDirIfNotExists(path: fs.PathLike) {
   if (!fs.existsSync(path)) {
     fs.mkdirSync(path, { recursive: true });
   }
 }
 
-export { LauncherFileSystem };
+function breakingVersion(version: string): string {
+  if (!semver.valid(version)) {
+    throw new Error('Version is not valid semver.');
+  }
+  if (semver.prerelease(version)) {
+    return version;
+  }
+  switch (semver.major(version)) {
+    case 0:
+      switch (semver.minor(version)) {
+        case 0:
+          return `0.0.${semver.patch(version)}`;
+        default:
+          return `0.${semver.minor(version)}.x`;
+      }
+    default:
+      return `${semver.major(version)}.x.x`;
+  }
+}
```

Comment:
how about?

`function breakingVersion(version) {
  if (!semver.valid(version)) {
    throw new Error('Version is not valid semver.');
  }

  if (semver.prerelease(version)) {
    return version;
  }

  const major = semver.major(version);
  const minor = semver.minor(version);
  const patch = semver.patch(version);

  if (major === 0) {
    return minor === 0 ? `0.0.${patch}` : `0.${minor}.x`;
  }

  return `${major}.x.x`;
}`